app/serializers/discourse_post_event/event_serializer.rb

```diff
@@ -129,8 +131,50 @@ def should_display_invitees
       (object.public? && object.invitees.count > 0) ||
         (object.private? && object.raw_invitees.count > 0)
     end
+
     def category_id
       object.post.topic.category_id
     end
+
+    def include_next_dates?
+      object.recurrence
+    end
+
+    def next_dates
+      how_many_recurring_events =
```

Comment:
I think it will be beneficial to extract this to the `RRuleGenerator.generate` method via an additional parameter, `max_years: N`. Then you could have an internal method there, `def how_many_recurring_events(max_years)` that does the calculation you already have here times `max_years`.

This way you remove complexity from the serializer and end up with:

```ruby
RRuleGenerator.generate(
  recurrence_rule,
  object.starts_at.in_time_zone(object.timezone),
  tzid: object.timezone,
  max_years: 1
).map { |date| { starts_at: date, ends_at: date + difference.seconds } }
```